Tasks/Gradle/sonarqube-common/sonarqube-qualitygate.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('../testlib/tasklib-wrapper');
+
+// Creates a build summary section to display the quality gate status.
+// Returns null if analysisStatus was null.
+export function createBuildSummaryQualityGateSection(analysisStatus:string):string {
+    if (!analysisStatus) {
+        return null;
+    }
+
+    var visualStatus:QualityGateVisualStatus = QualityGateVisualStatus.createFromAnalysisStatus(analysisStatus);
+
+    // ES6 template literal usage to streamline creating this section.
+    var reportContents:string  = `<div style="padding:5px 0px">
+        <span>Quality Gate</span>
+    <span style="padding:4px 10px; margin-left: 5px; background-color:${visualStatus.Color}; color:#fff; display:inline-block">${visualStatus.Label}</span>
+        </div>`;
+
+    return reportContents;
+}
+
+class QualityGateVisualStatus {
```

Comment:
This private class has a single method you are using so there is no point in it.